src/backend/joanie/core/api/admin/__init__.py

```diff
@@ -636,6 +641,37 @@ def generate_certificate(self, request, pk=None):  # pylint:disable=unused-argum
 
         return Response(certificate, status=HTTPStatus.CREATED)
 
+    @action(methods=["POST"], detail=True)
+    def cancel(self, request, pk=None):  # pylint:disable=unused-argument
+        """
+        Cancel an order.
+        """
+        order = self.get_object()
+
+        order.flow.cancel()
+
+        return Response(status=HTTPStatus.OK)
```

Comment:
This endpoint already exists. This is the `destroy` one. Sorry I didn't catch that before.